### ISSUE TITLE
Performance regression for pytorch_mobilenet_v3 on 20210317

### ISSUE THREAD

**Issue by xuzhao9** [pytorch/pytorch#54177]
## 🐛 Bug

On 20210317, pytorch_mobilenet_v3 performance on TorchBench regresses for 80%
Group A: benchmark-result-v0.1-20210316140129.json
Group B: benchmark-result-v0.1-20210317140247.json
```
Benchmark                                                 Run 20210316  Run 20210317  Delta
test_train[BERT_pytorch-cpu-jit]                          0.180699      0.181264     +0.311839%
test_eval[BERT_pytorch-cpu-jit]                           0.0445363     0.0444026    -0.301144%
test_train[BERT_pytorch-cpu-eager]                        0.186492      0.186447     -0.024502%
test_eval[BERT_pytorch-cpu-eager]                         0.0515908     0.0516071    +0.031570%
test_train[BERT_pytorch-cuda-jit]                         0.0852942     0.084899     -0.465524%
test_eval[BERT_pytorch-cuda-jit]                          0.0187054     0.0185803    -0.673232%
test_train[BERT_pytorch-cuda-eager]                       0.0874043     0.0873521    -0.059762%
test_eval[BERT_pytorch-cuda-eager]                        0.0211939     0.0211864    -0.035391%
test_train[Background_Matting-cuda-jit]                   5.03006       4.95509      -1.513151%
test_train[Background_Matting-cuda-eager]                 5.01496       5.00141      -0.270995%
test_train[LearningToPaint-cpu-jit]                       0.0345626     0.0343854    -0.515494%
test_eval[LearningToPaint-cpu-jit]                        0.00643133    0.00636817   -0.991789%
test_train[LearningToPaint-cpu-eager]                     0.034682      0.034521     -0.466434%
test_eval[LearningToPaint-cpu-eager]                      0.00661685    0.00662962   +0.192631%
test_train[LearningToPaint-cuda-jit]                      0.0157747     0.0158159    +0.260167%
test_eval[LearningToPaint-cuda-jit]                       0.00240728    0.00240807   +0.032830%
test_train[LearningToPaint-cuda-eager]                    0.0156815     0.01569      +0.054302%
test_eval[LearningToPaint-cuda-eager]                     0.00234721    0.00234158   -0.240429%
test_train[attention_is_all_you_need_pytorch-cpu-jit]     1.04345       1.07715      +3.128738%
test_eval[attention_is_all_you_need_pytorch-cpu-jit]      0.251182      0.256899     +2.225166%
test_train[attention_is_all_you_need_pytorch-cpu-eager]   0.960396      0.976198     +1.618675%
test_eval[attention_is_all_you_need_pytorch-cpu-eager]    0.260453      0.257726     -1.058230%
test_train[attention_is_all_you_need_pytorch-cuda-jit]    0.460141      0.459503     -0.138903%
test_eval[attention_is_all_you_need_pytorch-cuda-jit]     0.142123      0.142187     +0.045146%
test_train[attention_is_all_you_need_pytorch-cuda-eager]  0.433541      0.432873     -0.154280%
test_eval[attention_is_all_you_need_pytorch-cuda-eager]   0.141262      0.141368     +0.075030%
test_train[demucs-cpu-jit]                                0.352669      0.353448     +0.220454%
test_eval[demucs-cpu-jit]                                 0.0979476     0.0982247    +0.282067%
test_train[demucs-cpu-eager]                              0.367281      0.333403     -10.161033%*
test_eval[demucs-cpu-eager]                               0.101314      0.0988186    -2.525686%
test_train[demucs-cuda-jit]                               0.16872       0.168715     -0.003155%
test_eval[demucs-cuda-jit]                                0.0817282     0.0818301    +0.124473%
test_train[demucs-cuda-eager]                             0.168759      0.168665     -0.055491%
test_eval[demucs-cuda-eager]                              0.0817207     0.0818454    +0.152436%
test_train[dlrm-cpu-eager]                                0.00168901    0.00171276   +1.386618%
test_eval[dlrm-cpu-eager]                                 0.000552197   0.000556377  +0.751336%
test_train[dlrm-cuda-eager]                               0.00315873    0.00315554   -0.101161%
test_eval[dlrm-cuda-eager]                                0.000943121   0.000951918  +0.924078%
test_train[fastNLP-cpu-jit]                               0.00242984    0.00244216   +0.504092%
test_eval[fastNLP-cpu-jit]                                0.000203432   0.00020093   -1.245299%
test_train[fastNLP-cpu-eager]                             0.00239239    0.00240708   +0.610547%
test_eval[fastNLP-cpu-eager]                              0.000200417   0.0002035    +1.515442%
test_train[fastNLP-cuda-jit]                              0.0017709     0.00176924   -0.093664%
test_eval[fastNLP-cuda-jit]                               0.000180764   0.000180804  +0.022175%
test_train[fastNLP-cuda-eager]                            0.00165498    0.00165124   -0.226288%
test_eval[fastNLP-cuda-eager]                             0.000183508   0.000184523  +0.549703%
test_train[moco-cuda-jit]                                 0.891933      0.89177      -0.018321%
test_eval[moco-cuda-jit]                                  0.424216      0.424559     +0.080691%
test_train[moco-cuda-eager]                               0.891773      0.891663     -0.012365%
test_eval[moco-cuda-eager]                                0.424114      0.424726     +0.144164%
test_train[pytorch_mobilenet_v3-cpu-jit]                  0.195118      1.04195      +81.273774%*
test_eval[pytorch_mobilenet_v3-cpu-jit]                   0.0186275     0.10254      +81.833886%*
test_train[pytorch_mobilenet_v3-cpu-eager]                0.204235      1.05723      +80.682043%*
test_eval[pytorch_mobilenet_v3-cpu-eager]                 0.0217795     0.106492     +79.548295%*
test_train[pytorch_mobilenet_v3-cuda-jit]                 0.19641       1.04843      +81.266345%*
test_eval[pytorch_mobilenet_v3-cuda-jit]                  0.0187775     0.103831     +81.915378%*
test_train[pytorch_mobilenet_v3-cuda-eager]               0.205336      1.06389      +80.699536%*
test_eval[pytorch_mobilenet_v3-cuda-eager]                0.021818      0.107973     +79.793060%*
test_train[pytorch_struct-cpu-jit]                        1.52097       1.32595      -14.707910%*
test_eval[pytorch_struct-cpu-jit]                         0.00190022    0.00203652   +6.692904%*
test_train[pytorch_struct-cpu-eager]                      1.51775       1.32218      -14.791242%*
test_eval[pytorch_struct-cpu-eager]                       0.00209851    0.00225057   +6.756406%*
test_train[pytorch_struct-cuda-jit]                       0.241742      0.241826     +0.034544%
test_eval[pytorch_struct-cuda-jit]                        0.00126673    0.00125221   -1.159568%
test_train[pytorch_struct-cuda-eager]                     0.241841      0.241907     +0.027510%
test_eval[pytorch_struct-cuda-eager]                      0.00149828    0.00149975   +0.098253%
test_eval[yolov3-cpu-eager]                               0.636595      0.583661     -9.069328%*
test_train[yolov3-cuda-eager]                             9.99062      10.2197       +2.241218%
test_eval[yolov3-cuda-eager]                              0.605965      0.593679     -2.069483%
```

[Bisection job](https://github.com/pytorch/benchmark/actions/runs/661801888) shows [ce2f7183](https://github.com/pytorch/pytorch/commit/ce2f71836cc29ca499d2c26e9337aadee659fa94) is the root cause, parent commit is [f52a3bd6](https://github.com/pytorch/pytorch/commit/f52a3bd634ba3fca6de84e4dba4d8c13729a3bdc)

ABTesting in progress.

cc @VitalyFedyunin @ngimel

**Comment by ngimel** [comment]
That bisection is plausible, however, it was necessary because mkldnn produced wrong results for grouped convolutions. 
@VitalyFedyunin we don't really need it in the master, only in 1.8.1, right? In the master we only need test coverage, the new OneDNN version handles this case correctly.

**Comment by xuzhao9** [comment]
If that is the case, I guess we should revert the PR from master and cherry-pick it into 1.8.1.

**Comment by malfet** [comment]
@xuzhao9 are results of the test validated somehow in torchbench? I.e. does network accuracy stays the same before and after the change?

**Comment by wconstab** [comment]
no, we don't currently monitor correctness except in one case:  jit v eager is compared, and eager is assumed correct.

we could at least record expected output values for each model given the snapshot data input, so that a change in output can be detected.  this is far from the ideal of actually measuring training accuracy, but it could help.

**Comment by xuzhao9** [comment]
A fix on the master branch is open: https://github.com/pytorch/pytorch/pull/52897

**Comment by VitalyFedyunin** [comment]
I'm working on landing internal OneDNN update and disabling work-around dispatch

**Comment by xuzhao9** [comment]
After #54890 is merged, the benchmarking result has returned to normal.

**Comment by xuzhao9** [comment]
Here is the execution flow of the bug:
1. [The model](https://github.com/pytorch/benchmark/blob/master/torchbenchmark/models/pytorch_mobilenet_v3/mobilenetv3.py#L148) initializes a configuration matrix with exp value to be multiplies of 24.
2. The model defines a sequential operation, `self.conf` including convolutions with a group value of the exp value [here](https://github.com/pytorch/benchmark/blob/master/torchbenchmark/models/pytorch_mobilenet_v3/mobilenetv3.py#L96)
3. The `self.conv` operation is called during `forward()` [here](https://github.com/pytorch/benchmark/blob/master/torchbenchmark/models/pytorch_mobilenet_v3/mobilenetv3.py#L111)
